### DISCUSSION TITLE
Process for deciding who's on the front page?

### DISCUSSION THREAD

**Discussion by code-ghalib** [ocaml/ocaml.org#312] [Q&A]
I'm wondering whether there's a community process or guidelines for who gets included on the v3.ocaml.org front page.  

I think that part of the page serves as marketing mechanism for OCaml as a language/ecosystem - if you learn OCaml, you can land a job at X/Y/Z company.  

It works the other way too of course, if you want to work in OCaml, come work for X/Y/Z company. 

Is there already a process for how it is decided who appears on the front page, and who appears in the industrial users section?

We can as a community decide, that since this is high-visibility web-estate, it be auctioned, or be limited to OCaml Software Foundation members, or something else.  Or we can agree to enhance the front page to show random industrial users every time. 

I'm creating this issue for discussion purposes and seeking proposals and feedback from the community.

What I'd like to avoid is arbitrary inclusions/removals, for example, [like this](https://github.com/ocaml/v3.ocaml.org-server/commit/833127803144d7582d52d8010b73164f6b1d618d).

**Comment by tmattio** [comment]
Thanks @code-ghalib, that's a very important question 🙂 

We don't have a specific process to decide which logos are included on the homepage. The general governance of the website is described at https://v3.ocaml.org/governance though, and that should give some ideas on how decisions are taken.

Even though we're getting close to the launch date, the website is still in preview at the moment, so the decision process is a bit different now than it ought to be once it's officially launched. To have a reasonable momentum during development, we're "queuing" changes by pushing them to the preview site. The owners and delegates (as described in the governance document above) will be giving the green light for a proposed version, and we will launch this approved version of the site.

Now, once the website is launched, I expect that the decision process will be a bit different, and the owners and delegates should validate important changes such as the logos on the front page.

So to be a bit more specific, the owners and delegates will approve the currently proposed list of logos when giving the green light for launch and will be able to review PRs that propose changes to these once it's launched.

Now, this governance structure describes who's responsible for the final decision, but of course, OCaml.org remains a community website, so anyone can open PRs or suggest changes. If you think the logo cloud is not adequate or relevant, your suggestions are very much welcome!

I hope this answers the question and addresses the (rightful!) concerns you're expressing 🙂

**Reply by code-ghalib** [reply]
Thank you @tmattio for your detailed answer - it is great to see that there is a documented governance process.  

If I may summarize my understanding of the document as it pertains to this question: 
- owners or delegates (currently @avsm) make the final decisions on the content based on "rough consensus"
- contributions from the community (even people like me!) are welcome

Is this accurate?  

If so, I would like to propose that we add back the Bloomberg logo - that would address my immediate concern.  

In the longer term though, I'm not sure how this scales or stays "equitable". I am for example surprised to not see LexiFi show up on the front page.  

(Also, as an aside, the Facebook folks probably want the Meta logo now? I can add that PR if you want)

If we do intend to keep a static list of some `n` industry leaders that trust OCaml, I think it would be nice to have some general idea from @avsm for qualitative/quantitative criterion for inclusion/exclusion to evaluate PRs/proposals to add/remove logos.

Some example criteria may be:
- size of industry they lead
- market cap / valuation
- size of code-base in OCaml
- code/content contributions to the community
- financial contributions to OCaml community
- some other metric I haven't thought of

I am slightly more in favor of a different approach though, which is to flash logos of a randomly selected `n` industrial users (and even academic users!), and change it every 3 seconds or so.  

I feel it is:
- more equitable
- gives a better idea of the wide range of applications and trust shown in OCaml as a language

Of course I understand you're preparing for launch and have no time to entertain such wild ideas right now - but I am willing to finally try to learn the react / ReScript necessary to do this, if there is agreement from folks watching this space that this would be a desirable solution.

**Reply by tmattio** [reply]
Copying @avsm answer below, I think it was intended as a reply to this thread:

> > contributions from the community (even people like me!) are welcome
> 
> This is by far the biggest criteria for inclusion: engagement :-) Contributions from you are extremely welcome, and the essential gateway for inclusion on the site for a someone is that someone sends in the necessary changes. Where something is clearly out of scope, the maintainers may step in, but we otherwise take a broad view of what can get onto the front page.

**Reply by code-ghalib** [reply]
@avsm @tmattio just an fyi in case you missed it, I created [this PR](https://github.com/ocaml/ocaml.org/pull/319).

**Comment by kohlivarun5** [comment]
It might be worth tying it back to the Software Foundation aswell. I don't see any mention of OSF on the front page.

**Reply by avsm** [reply]
@code-ghalib:
> contributions from the community (even people like me!) are welcome

This is by far the biggest criteria for inclusion: engagement :-) Contributions from you are extremely welcome, and the essential gateway for inclusion on the site for a someone is that someone sends in the necessary changes.  Where something is clearly out of scope, the maintainers may step in, but we otherwise take a broad view of what can get onto the front page.
